skopt/tests/test_space.py

```diff
@@ -65,22 +65,40 @@ def test_integer():
     assert_array_equal(a.inverse_transform(random_values), random_values)
 
 
+def test_categorical_transform_binary():
+    categories = ["apple", "orange"]
+    cat = Categorical(categories)
+
+    apple = [0.]
+    orange = [1.]
+
+    assert_equal(cat.transform(["apple"]).size, cat.transformed_size)
+    assert_array_equal(cat.transform(categories), [apple, orange])
```

Comment:
why would we purposefully take up more memory when this is completely transparent to the user, is anyone supposed to directly use the `warped` space?